workflow: Remove `yarn.lock` as it no longer exists

nathanstilwell opened the pull request:
Removed the `yarn.lock` file path from cluster release next action, as it has been removed since we swithed to pnpm.

Epic: none

Release note: None

Commit:
workflow: Remove `yarn.lock` as it no longer exists

Removed the `yarn.lock` file path from cluster release next action, as
it has been removed since we swithed to pnpm.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107884)
<!-- Reviewable:end -->

Commit:
ui: Clean up .yarnrc

Yarn has been replaced by PNPM so this rc file is no longer necessary.

Epic: none

Release note: None

sjbarag reviewed (approved):
Oof good catch on the yarnrc!

nathanstilwell commented:
bors r+
